trl/trainer/grpo_trainer.py

```diff
@@ -699,11 +803,19 @@ def _generate_and_score_completions(
 
             # Generate completions using vLLM: gather all prompts and use them in a single call in the main process
             all_prompts_text = gather_object(prompts_text)
+            all_images = gather_object(images) if images is not None else None
             if self.accelerator.is_main_process:
                 # Since 'prompts' contains 'num_generations' duplicates, we first take unique prompts, and generate
                 # num_generations outputs for each one. This is faster than generating outputs for each duplicate
                 # prompt individually.
                 ordered_set_of_prompts = all_prompts_text[:: self.num_generations]
+                ordered_set_of_images = all_images[:: self.num_generations] if all_images is not None else None
+                # cf. https://docs.vllm.ai/en/stable/serving/multimodal_inputs.html#image
+                if ordered_set_of_images is not None:
+                    ordered_set_of_prompts = [
+                        {"multi_modal_data": {"image": image}, "prompt": prompt}
+                        for image, prompt in zip(ordered_set_of_images, ordered_set_of_prompts)
+                    ]
                 with profiling_context(self, "vLLM.generate"):
                     completion_ids = self.vllm_client.generate(
```

Comment:
The changes in https://github.com/huggingface/trl/commit/f713f614e9df6b1f581c38724cb6b92da8edac62 imply that it's now not possible to send the multi_modal_data without adapting the new vllm server. I think the following would have to change:

- https://github.com/huggingface/trl/blob/26d86757a7c7e24e397ea44f57ecce6031dfac01/trl/extras/vllm_client.py#L132
- https://github.com/huggingface/trl/blob/26d86757a7c7e24e397ea44f57ecce6031dfac01/trl/scripts/vllm_serve.py#L316